typescript/sdk/src/ism/EvmIsmModule.ts

```diff
@@ -1,89 +1,727 @@
+          to: this.args.addresses.deployedIsm,
+          data: Ownable__factory.createInterface().encodeFunctionData(
+            'transferOwnership(address)',
+            [targetOwner],
+          ),
+        });
+        return [tx];
+      }
+      // no owner diff == no tx to return
+      return [];
+    }
+    // else if it's a routing ISM - update the existing one
+
+    // only check for mailbox diff if it's a fallback routing ISM
+    if (targetIsmType === IsmType.FALLBACK_ROUTING) {
+      // can only retreive mailbox address if current ISM type is also Fallback Routing
+      const mailboxAddress =
+        currentConfig.type === IsmType.FALLBACK_ROUTING
+          ? await MailboxClient__factory.connect(
+              this.args.addresses.deployedIsm,
+              provider,
+            ).mailbox()
+          : ''; // empty string to force a mailbox diff
+
+      // if mailbox delta, deploy new routing ISM before updating
+      // this will always be the case if the current ISM is not a fallback routing ISM
+      if (!eqAddress(mailboxAddress, this.args.addresses.mailbox)) {
+        const newIsm = await this.deployRoutingIsm({
+          config: targetConfig,
+          logger,
+        });
+
+        this.args.addresses.deployedIsm = newIsm.address;
+      }
+    }
+
+    // this calls resolveOrDeployAccountOwner under the hood
+    const delta = await routingModuleDelta(
+      this.chainName,
+      this.args.addresses.deployedIsm,
+      targetConfig,
+      this.multiProvider,
+      this.factories,
+      this.args.addresses.mailbox,
+    );
+
+    return await this.updateRoutingIsm({
+      delta,
+      config: targetConfig,
+      logger,
     });
-    return [];
   }
 
   // manually write static create function
-  public static async create({
-    chain,
-    config,
-    deployer,
-    factories,
-    multiProvider,
-  }: {
+  public static async create(params: {
     chain: ChainNameOrId;
     config: IsmConfig;
     deployer: HyperlaneDeployer<any, any>;
-    factories: HyperlaneContracts<ProxyFactoryFactories>;
+    factories: HyperlaneAddresses<ProxyFactoryFactories>;
+    mailbox: Address;
     multiProvider: MultiProvider;
   }): Promise<EvmIsmModule> {
-    const destination = multiProvider.getChainName(chain);
-    const ismCreator = new EvmIsmCreator(deployer, multiProvider, factories);
-    const deployedIsm = await ismCreator.deploy({
-      config,
-      destination,
-    });
-    return new EvmIsmModule(multiProvider, deployer, {
+    const { chain, config, deployer, factories, mailbox, multiProvider } =
+      params;
+    const module = new EvmIsmModule(multiProvider, deployer, {
       addresses: {
         ...factories,
-        deployedIsm: deployedIsm.address,
+        mailbox,
+        deployedIsm: ethers.constants.AddressZero,
       },
       chain,
       config,
     });
+    const deployedIsm = await module.deploy({ config });
+    module.args.addresses.deployedIsm = deployedIsm.address;
```

Comment:
I find this slightly odd
why wouldn't module.deploy assign to this?

Reply:
scratch the above - the main reason is because we should be able to reuse the .deploy function to deploy submodules without overwriting the top-level ism address

Reply:
interesting, okay